### PR TITLE
Adding doctests to task 1 and changing to ndenumerate

### PR DESCRIPTION
**Pull request by jsphconnell**
Hi @Enaitz - Added in doctests for task1 - handles https://github.com/jsphconnell/ARC/issues/8.
We can close that issue now :)

Also rewrote the code to use ndenumerate to keep common across files.

### CODE REVIEW

File: src/solution_0ca9ddb6.py [hunk 1]
```diff
@@ -34,10 +47,16 @@ def solve(input_grid):
...
+    array([[4, 0, 4],
+           [0, 2, 0],
+           [4, 0, 4]])
     """
     numpy_grid = np.array(input_grid)
-    
-    for ix, iy in np.ndindex(numpy_grid.shape):
+
+    for (ix, iy), value in np.ndenumerate(numpy_grid):
         if numpy_grid[ix, iy] == 1 or numpy_grid[ix, iy] == 2:
```
Comment (pedrera-x) [inline src/solution_0ca9ddb6.py]: consider changing if numpy_grid[ix, iy] == 1 or numpy_grid[ix, iy] == 2: to if value in [1, 2]: (at least use value as you have it there)

**Review by pedrera-x — DISMISSED** [review]
made a small comment as consideration

File: src/solution_0ca9ddb6.py (hunk 1 above)
Comment (pedrera-x) [reply src/solution_0ca9ddb6.py]: @jsphconnell I changed the code - thought about it further and I think makes sense

Comment (jsphconnell) [reply src/solution_0ca9ddb6.py]: @Enaitz Looks good - a great suggestion. Tested it locally and it still works. Lets merge this in, can you approve and merge as i opened PR?

**Review by pedrera-x — APPROVED** [review]

**Pull request merged**